### DISCUSSION TITLE
Rendering of high unicode characters in syntax highlighting is broken on Firefox

### DISCUSSION THREAD

**Discussion by digama0** [community/community#103098] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Here's a screenshot of https://github.com/leanprover-community/mathlib4/blob/aa20ff6f3effa95af3be5043c93d35a712cd79c4/Mathlib/Analysis/Convex/Cone/Pointed.lean#L25-L26 on firefox, on a line which contains the high-unicode character `𝕜`:
![image](https://github.com/community/community/assets/868588/1fa819e6-007e-48a2-9a83-071bd77ccc17)

The line appears to have been broken into two-UTF-16 codepoint chunks, which causes `𝕜` (which is larger than 0x10000 and hence needs to be represented as two UTF-16 characters) to be split down the middle in some but not all occurrences (you can see a correctly rendered `𝕜` earlier on the same line). As far as I can tell this is not the fault of the syntax highlighter, which suggests to highlight none of the text in question.

The correct text appears briefly during page load, and the text in the textarea (i.e. what is copied) is also correct.

Here is the same line in Chromium, which does not split the span:
![image](https://github.com/community/community/assets/868588/a6a8d431-e017-4f7d-b9a5-1cfc12e72f5d)

**Comment by Cbodfield** [comment, marked as answer]
Thanks for the bug report @digama0! 

We made some changes recently that fixed this issue and also improve rendering performance. Once your browser is updated to Firefox 124 (the current latest version) these will be fixed for you.

**Comment by fpvandoorn** [comment]
Thanks! I can confirm it looks correct for me on Firefox 124.
